pkg/cmd/pulumi/package_add.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+	"github.com/spf13/cobra"
+)
+
+// Constructs the `pulumi package add` command.
+func newPackageAddCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add <provider> [provider parameters]",
+		Args:  cobra.MinimumNArgs(1),
+		Short: "Add a package to your Pulumi project",
+		Long:  "Add a package to your Pulumi project.",
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			proj, root, err := readProject()
+			if err != nil && errors.Is(err, workspace.ErrProjectNotFound) {
+				return errors.New("pulumi package add must be run in a Pulumi project")
```

Comment:
Note: I think `-C` and `--cwd` will work automatically for this command. I think you can just return the original `err` from `readProject` here, that is what we seem to do in another commands.